controller/api/authentication.py

```diff
@@ -9,3 +10,15 @@ def authenticate(self, request):
         Authenticate the request for anyone!
         """
         return AnonymousUser(), None
+
+
+class AnonymousOrAuthenticatedAuthentication(authentication.BaseAuthentication):
+
+    def authenticate(self, request):
+        """
+        Authenticate the request for anyone or if a valid token is provided, a user.
+        """
+        try:
+            return TokenAuthentication.authenticate(TokenAuthentication(), request)
```

Comment:
Hmm, this might not work when LDAP auth is enabled. We should support that somehow.